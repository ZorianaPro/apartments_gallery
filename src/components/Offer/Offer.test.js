import React from 'react';
import Offers from '../Offer';
import {  mount } from 'enzyme';

describe('Offers', () => {

	let wrapper, props;

	beforeEach(() => {
		props = {id: 123, details: {name: 'TEST'}, photos: [{m: 'qwer'}], price: 123, location: {name: 'Test'}};
		wrapper = mount(<Offers {...props}/>);
	});

	test('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});

	describe('no photos', () => {

		let wrapper, props;

		beforeEach(() => {
			props = {id: 123, details: {name: 'TEST'}, photos: [{}], price: 123, location: {name: 'Test'}};
			wrapper = mount(<Offers {...props}/>);
		});

		test('renders default picture', () => {
			expect(wrapper.find('img').prop('src')).toBe('default-picture');
		});

	});
});

