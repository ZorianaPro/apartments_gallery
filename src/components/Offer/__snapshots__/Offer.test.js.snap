// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offers renders as expected 1`] = `
<Offer
  details={
    Object {
      "name": "TEST",
    }
  }
  id={123}
  location={
    Object {
      "name": "Test",
    }
  }
  photos={
    Array [
      Object {
        "m": "qwer",
      },
    ]
  }
  price={123}
>
  <div
    className="offer-item"
    key="123"
  >
    <div
      className="image-container"
    >
      <img
        alt="TEST"
        className="offer-image"
        height="100%"
        src="qwer"
        width="100%"
      />
    </div>
    <div
      className="offer-item-info"
    >
      <div
        className="info-container"
      >
        <div
          className="offer-title cut-title"
        >
          TEST
        </div>
        <div>
          Test
        </div>
      </div>
      <div
        className="price-container"
      >
        <span>
           
          /night
        </span>
      </div>
    </div>
  </div>
</Offer>
`;
